File: src/main/java/me/zeroeightsix/kami/event/events/GuiScreenEvent.java
package me.zeroeightsix.kami.event.events;

import net.minecraft.client.gui.GuiScreen;

/**
 * Created by 086 on 17/11/2017.
 */
public class GuiScreenEvent {

    private GuiScreen screen;

    public GuiScreenEvent(GuiScreen screen) {
        super();
        this.screen = screen;
    }

    public GuiScreen getScreen() {
        return screen;
    }

    public void setScreen(GuiScreen screen) {
        this.screen = screen;
    }

    public static class Displayed extends GuiScreenEvent {
        public Displayed(GuiScreen screen) {
            super(screen);
        }
    }

    public static class Closed extends GuiScreenEvent {
        public Closed(GuiScreen screen) {
            super(screen);
        }
    }

}

File: src/main/java/me/zeroeightsix/kami/event/events/GuiScreenEvent.kt
package me.zeroeightsix.kami.event.events

import net.minecraft.client.gui.GuiScreen

/**
 * Created by 086 on 17/11/2017.
 * Updated by Xiaro on 18/08/20
 */
open class GuiScreenEvent(var screen: GuiScreen?) {

    class Displayed(screen: GuiScreen?) : GuiScreenEvent(screen)
    class Closed(screen: GuiScreen?) : GuiScreenEvent(screen)
}